### PR TITLE
Make comments polymorphic

### LINKED ISSUE #346: Refactoring: Make Comment polymorphic
The Comment model has grown a set of commentable records' foreign keys. We should refactor it into a `belongs_to :commentable, polymorphic: true` instead of listing all types of recods individually.

There's no deadline for this and can be done whenever there's time. @marcgreenstock and @lucaspinto, is this something you could see yourself working on?

### PR DESCRIPTION
**Pull request by carpodaster**
Closes #346

### CODE REVIEW

**Comment by F3PiX** [comment]
👏  👏

**Comment by alicetragedy** [comment]
🎉 🎊

**Comment by carpodaster** [comment]
@klappradla @michaelem since you gave me the excellent tips about data transformations in schema migrations, I'd really appreciate a looksie :)

File: spec/factories/comment.rb [hunk 1]
```diff
@@ -4,11 +4,11 @@
     text { FFaker::CheesyLingo.paragraph }
 
     factory :team_comment do
-      team
+      commentable { create :team }
```
Comment (klappradla) [inline spec/factories/comment.rb]: The `association` syntax used to be broken for polymorphic, but afaik on the current FactoryGirl version you can now use:

```ruby
association :commentable, factory: :team
````

But comes down to personal preference in the end @carpodaster 😉

File: db/migrate/20160712095859_make_existing_foreign_keys_polymorphic_for_comments.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  belongs_to :team
+  belongs_to :application
+  belongs_to :project
+end
+
+class MakeExistingForeignKeysPolymorphicForComments < ActiveRecord::Migration
+  def up
+    ActiveRecord::Base.transaction do
+      Comment.includes(:team, :application, :project).find_each do |comment|
+        if assoc = [:team, :application, :project].detect { |a| comment.send(a).present? }
```
Comment (klappradla) [inline db/migrate/20160712095859_make_existing_foreign_keys_polymorphic_for_comments.rb]: Awesome approach using `detect` here 🎉

Comment (klappradla) [inline db/migrate/20160712095859_make_existing_foreign_keys_polymorphic_for_comments.rb]: *(accidentally deleted my comment again I guess..)*
Awesome approach using `detect` here I wanted to say 🎉

File: spec/mailers/project_mailer_spec.rb [hunk 3]
```diff
@@ -20,8 +20,8 @@
   end
 
   describe '#comment' do
-    let!(:comment) { create :comment, project: project }
-    let!(:commenter) { create(:comment, project: project).user }
+    let!(:comment) { create :comment, commentable: project }
+    let!(:commenter) { create(:comment, commentable: project).user }
```
Comment (klappradla) [inline spec/mailers/project_mailer_spec.rb]: Maybe I don't get it, but I don't see the point in creating two comment records here. Wouldn't this be enough?

```ruby
let!(:comment) { create :comment, commentable: project }
let(:commenter) { comment.user }

Comment (carpodaster) [inline spec/mailers/project_mailer_spec.rb]: Totally! I was just changing the factories to get the tests pass, didn't look at the semantics. Will change it!

File: spec/factories/comment.rb (hunk 1 above)
Comment (carpodaster) [inline spec/factories/comment.rb]: 👍

**Pull request merged**